File: app/src/main/java/com/kunzisoft/keepass/settings/preference/InputTextExplanationPreference.java
package com.kunzisoft.keepass.settings.preference;

import android.content.Context;
import android.content.res.TypedArray;
import android.support.v7.preference.DialogPreference;
import android.util.AttributeSet;

import tech.jgross.keepass.R;

public class InputTextExplanationPreference extends DialogPreference {

    protected String explanation;

    public InputTextExplanationPreference(Context context) {
        this(context, null);
    }

    public InputTextExplanationPreference(Context context, AttributeSet attrs) {
        this(context, attrs, R.attr.dialogPreferenceStyle);
    }

    public InputTextExplanationPreference(Context context, AttributeSet attrs, int defStyleAttr) {
        this(context, attrs, defStyleAttr, defStyleAttr);
    }

    public InputTextExplanationPreference(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        TypedArray a = context.getTheme().obtainStyledAttributes(
                attrs,
                R.styleable.explanationDialog,
                0, 0);
        try {
            setExplanation(a.getString(R.styleable.explanationDialog_explanations));
        } finally {
            a.recycle();
        }
    }

    @Override
    public int getDialogLayoutResource() {
        return R.layout.pref_dialog_input_text_explanation;
    }

    public String getExplanation() {
        return explanation;
    }

    public void setExplanation(String explanation) {
        this.explanation = explanation;
    }
}

File: app/src/main/java/com/kunzisoft/keepass/settings/preference/InputTextExplanationPreference.kt
package com.kunzisoft.keepass.settings.preference

import android.content.Context
import android.support.v7.preference.DialogPreference
import android.util.AttributeSet

import com.kunzisoft.keepass.R

open class InputTextExplanationPreference @JvmOverloads constructor(context: Context,
                                                                    attrs: AttributeSet? = null,
                                                                    defStyleAttr: Int = R.attr.dialogPreferenceStyle,
                                                                    defStyleRes: Int = defStyleAttr)
    : DialogPreference(context, attrs, defStyleAttr, defStyleRes) {

    var explanation: String? = null

    init {
        val styleAttributes = context.theme.obtainStyledAttributes(
                attrs,
                R.styleable.explanationDialog,
                0, 0)
        try {
            explanation = styleAttributes.getString(R.styleable.explanationDialog_explanations)
        } finally {
            styleAttributes.recycle()
        }
    }

    override fun getDialogLayoutResource(): Int {
        return R.layout.pref_dialog_input_text_explanation
    }
}
